pkg/gatewayserver/redis/registry.go

```diff
@@ -32,36 +31,37 @@ type GatewayConnectionStatsRegistry struct {
 }
 
 var (
-	down        = "down"
-	up          = "up"
-	status      = "status"
-	errNotFound = errors.DefineNotFound("stats_not_found", "Gateway Stats not found")
+	down            = "down"
+	up              = "up"
+	status          = "status"
+	errNotFound     = errors.DefineNotFound("stats_not_found", "gateway stats not found")
+	errInvalidStats = errors.DefineCorruption("invalid_stats", "invalid `{type}` stats in store")
 )
 
 func (r *GatewayConnectionStatsRegistry) key(which string, uid string) string {
 	return r.Redis.Key(which, "uid", uid)
 }
 
 // Set sets or clears the connection stats for a gateway.
-func (r *GatewayConnectionStatsRegistry) Set(ctx context.Context, ids ttnpb.GatewayIdentifiers, stats *ttnpb.GatewayConnectionStats, update io.Traffic) error {
+func (r *GatewayConnectionStatsRegistry) Set(ctx context.Context, ids ttnpb.GatewayIdentifiers, stats *ttnpb.GatewayConnectionStats, updateUp bool, updateDown bool, updateStatus bool) error {
 	uid := unique.ID(ctx, ids)
 
 	defer trace.StartRegion(ctx, "set gateway connection stats").End()
 
 	var err error
 	if stats == nil {
 		_, err = r.Redis.Pipelined(func(p redis.Pipeliner) error {
-			if update.Up {
+			if updateUp {
 				if err := p.Del(r.key(up, uid)).Err(); err != nil {
 					return err
 				}
 			}
-			if update.Down {
+			if updateDown {
 				if err := p.Del(r.key(down, uid)).Err(); err != nil {
 					return err
 				}
 			}
-			if update.Status || (!update.Up && !update.Down) {
+			if updateStatus || (!updateUp && !updateDown) {
```

Comment:
Hmm, what is the use case for this?

Reply:
Gateway will not be shown as connected before any actual traffic occurs.

Reply:
Indeed. If the gateway is connected, it's online